webapp/channels/src/components/data_prefetch/index.ts

```diff
@@ -39,39 +37,56 @@ enum Priority {
 // function to return a queue obj with priotiy as key and array of channelIds as values.
 // high priority has channels with mentions
 // medium priority has channels with unreads
+// <10 unread channels. Prefetch everything.
+// 10-20 unread. Prefetch only mentions, capped to 10.
+// >20 unread. Don't prefetch anything.
 const prefetchQueue = memoizeResult((
     unreadChannels: Channel[],
     memberships: RelationOneToOne<Channel, ChannelMembership>,
     collapsedThreads: boolean,
 ) => {
-    return unreadChannels.reduce((acc: Record<string, string[]>, channel: Channel) => {
+    const unreadChannelsCount = unreadChannels.length;
+    let defaultResult: {
+        1: string[];
+        2: string[];
+        3: string[];
+    } = {
+        [Priority.high]: [], // 1 being high priority requests
+        [Priority.medium]: [],
+        [Priority.low]: [], //TODO: add chanenls such as fav.
+    };
+    if (!unreadChannelsCount || unreadChannelsCount > 20) {
+        return defaultResult;
+    }
+    for (const channel of unreadChannels) {
         const channelId = channel.id;
         const membership = memberships[channelId];
 
+        if (unreadChannelsCount >= 10 && defaultResult[Priority.high].length >= 10) {
```

Comment:
Would you mind adding constants for the 10/20? It'll make it easier to update those if we want to tweak them a bit